src/microvm/src/vmm/microvm/mod.rs

```diff
@@ -236,8 +241,19 @@ impl Vmm {
             }
         });
 
+        // We use an atomic to pass the id of the created thread back to the caller context. We
+        // need this because std::thread's JoinHandle does not expose the tid.
+        let pthread_id_holder: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
+
         let microvm_clone: Arc<Mutex<MicroVm>> = microvm.clone();
+        let pthread_id_holder_clone: Arc<AtomicUsize> = pthread_id_holder.clone();
         let vcpu_thread: JoinHandle<Result<u16>> = std::thread::spawn(move || {
+            // Store the tid so that the caller can send signals to the vCPU thread.
+            // SAFETY: we are calling pthread_self() right after creating the thread so this is
+            // safe.
+            let pthread_id: libc::pthread_t = unsafe { pthread_self() };
+            pthread_id_holder_clone.store(pthread_id as usize, Ordering::Relaxed);
```

Comment:
There will be a race condition if an interrupt handler is sent before this is reached. We should have a barrier here.